src/i965_drv_video.c

```diff
@@ -2532,11 +2532,11 @@ i965_create_buffer_internal(VADriverContextP ctx,
         assert(buffer_store->buffer);
 
         if (!wrapper_flag) {
+            memset(buffer_store->buffer, 0, msize * num_elements);
             if (data)
                 memcpy(buffer_store->buffer, data, size * num_elements);
-            else
-                memset(buffer_store->buffer, 0, size * num_elements);
-        }
+        } else
+            memset(buffer_store->buffer, 0, 4);
```

Comment:
The allocated four bytes are meaningless when the buffer is allocated with wrapper flag. 
So it doesn't matter that it is not initialized to zero.

At the same time I don't think that it is necessary to initialize the buffer again when it needs to copy the passed data.
When the data_pointer is passed, the data length is consistent with the total size of allocated buffer.  
So I think that this patch is not needed.